src/test/ui-fulldeps/proc-macro/auxiliary/span-preservation.rs

```diff
@@ -0,0 +1,77 @@
+// no-prefer-dynamic
+
+#![crate_type = "proc-macro"]
+#![feature(proc_macro_diagnostic)]
+#![feature(proc_macro_hygiene)]
+#![feature(proc_macro_quote)]
+#![feature(proc_macro_span)]
```

Comment:
Would it be possible to write this test with only stable proc macros perhaps? I've found the best way to do that is just force tokens to round trip through a procedural macro but make sure the tokens have a compiler warning in them (like an unused `mut`), and that way we won't have to update this over time ideally

Reply:
I did try that (round-tripping was my first attempt), but I couldn't get the tests to work as I'd expect. I can try again, however.